### DISCUSSION TITLE
OPL3BankEditor on M1 Mac to drive OPL2 Board.

### DISCUSSION THREAD

**Discussion by cstanke** [Wohlstand/OPL3BankEditor#192] [Q&A] [answered]
Hello,

Can you tell me if the OPL3BankEditor works on M1 (Apple Silicon) Macs? Also can you tell me if the serial output I pasted below is expected? Thanks for any insight you can offer!

I compiled OPL3BankEditor with [MacPorts QT5](https://ports.macports.org/port/qt5/) installed on Apple M1 (aarch64) processor running macOS Monterey 12.6. I used ```qtmake``` and it compiled fine. It runs but it doesn't drive the OPL2 board. The same happens with the [released 1.5.1 version](https://github.com/Wohlstand/OPL3BankEditor/releases/download/v1.5.1/opl3_bank_editor-v1.5.1-macos.zip) of OPL3BankEditor.

Compilation warns that parts of the QT5 framework were compiled against MacOSX11.sdk and not the current one I have (MacOSX12.sdk) but I suppress the warning with ```CONFIG+=sdk_no_version_check```. The actual warnings come out of ```ld``` and I wonder if this warning might be part of the problem:
```
ld: warning: dylib (/opt/local/libexec/qt5/lib/QtSerialPort.framework/QtSerialPort) was built for newer macOS version (12.0) than being linked (11.0)
```

I sent OPL3BankEditor's output to ```screen``` as a serial monitor (via ```ttyp9``` to ```ptyp9``` and verified that OPL3BankEditor is outputting data. When I change instruments or any other settings I get the same output no matter what I change:

```
0000000    005660  037500  037503  005661  037501  037504  005662  037502
0000010    037505  005663  037510  037513  005664  037511  037514  005665
0000020    037512  037515  005666  037520  037523  005667  037521  037524
0000030    005670  037522  037525  000275  037520  037523  000275  037524
0000040    000275  037522  000275  037525  000275  037521  000275  005660
0000050    037500  037503  005661  037501  037504  005662  037502  037505
0000060    005663  037510  037513  005664  037511  037514  005665  037512
0000070    037515  005666  037520  037523  005667  037521  037524  005670
0000080    037522  037525  000275  037520  037523  000275  037524  000275
0000090    037522  000275  037525  000275  037521  000275  000040  000043
00000a0    177540  000143  177600  000203  000340  000343  000300  037500
00000b0    037503  000041  000044  177541  000144  177601  000204  000341
00000c0    000344  000301  037501  037504  000042  000045  177542  000145
00000d0    177602  000205  000342  000345  000302  037502  037505  000050
00000e0    000053  177550  000153  177610  000213  000350  000353  000303
00000f0    037510  037513  000051  000054  177551  000154  177611  000214
0000100    000351  000354  000304  037511  037514  000052  000055  177552
0000110    000155  177612  000215  000352  000355  000305  037512  037515
0000120    000060  000063  177560  000163  177620  000223  000360  000363
0000130    000306  037520  037523  000061  000064  177561  000164  177621
0000140    000224  000361  000364  000307  037521  037524  000062  000065
0000150    177562  000165  177622  000225  000362  000365  000310  037522
0000160    037525  000060  000063  177560  000163  177620  000223  000360
0000170    000363  000306  037520  037523  000064  000164  000224  000364
0000180    000307  037524  000062  177562  177622  000362  000310  037522
0000190    000065  000165  000225  000365  037525  000061  177561  177621
00001a0    000361  037521                                                
00001a4
```

When I simply press the '**C**' key (on the keyboard) to play a note, I get different output each time as the operator iterates from 0 to 8.

**OP-0**
```
0000000    000040  000043  000140  000143  170200  170203  000340  000343
0000010    030300  037500  037503  000040  000043  062640  025660  005660
0000020
```

**OP-1**
```
0000000    000041  000044  000141  000144  170201  170204  000341  000344
0000010    030301  037501  037504  000041  000044  062641  025661  005661
0000020
```

**OP-2**
```
0000000    000042  000045  000142  000145  170202  170205  000342  000345
0000010    030302  037502  037505  000042  000045  062642  025662  005662
0000020
```

**OP-3***
```
0000000    000050  000053  000150  000153  170210  170213  000350  000353
0000010    030303  037510  037513  000050  000053  062643  025663  005663
0000020
```

**OP-4**
```
0000000    000051  000054  000151  000154  170211  170214  000351  000354
0000010    030304  037511  037514  000051  000054  062644  025664  005664
0000020
```

**OP-5**
```
0000000    000052  000055  000152  000155  170212  170215  000352  000355
0000010    030305  037512  037515  000052  000055  062645  025665  005665
0000020
```

**OP-6**
```
0000000    000060  000063  000160  000163  170220  170223  000360  000363
0000010    030306  037520  037523  000060  000063  062646  025666  005666
0000020
```

**OP-7**
```
0000000    000061  000064  000161  000164  170221  170224  000361  000364
0000010    030307  037521  037524  000061  000064  062647  025667  005667
0000020
```

**OP-8**
```
0000000    000062  000065  000162  000165  170222  170225  000362  000365
0000010    030310  037522  037525  000062  000065  062650  025670  005670
0000020
```

So, is this expected output from OPL3BankEditor? I don't have another system to verify on.

Thanks,
-C.

**Comment by Wohlstand** [comment, marked as answer]
Ideally, it should work on any platform with any processor, I don't use any assembly specific to hardware. Anyway, the QtSerialPort library is significantly essential to communicate hardware. You could override the macOS minimal version at the side of my app to ensure all stuff is linked properly :eyes:

**Reply by cstanke** [reply]
Hello!

Well, thanks for the reply. Thanks also for building OPL3 Bank Editor and making it available to the public. I appreciate the effort and dedication this kind of thing takes and am very thankful for your work!

There was certainly something weird with the MacPorts install I had (probably because it was pulled over from multiple previous OS installs). Finally I just built Qt5 and Qt5SerialPort from source, built OPL3BankEditor on top of those, loaded a preset from the examples and everything worked great!

Thanks!
-C.
